pyro/distributions/multivariate_cauchy.py

```diff
@@ -0,0 +1,39 @@
+import torch
+from torch.distributions import constraints
+from pyro.distributions.transforms import LowerCholeskyAffine
+from pyro.distributions.torch import TransformedDistribution, Cauchy
+
+
+class MultivariateCauchy(TransformedDistribution):
+    r"""
+    Creates a multivariate Cauchy distribution parameterized by
+    :attr:`loc` and :attr:`scale_tril` where::
+
+        X ~ Cauchy(loc, 1.0)
+        Y = scale_tril @ X ~ MultivariateCauchy(loc, scale_tril)
+
+    :param torch.tensor loc: the location parameter of the distribution, a D-dimensional vector.
+    :param torch.tensor scale_tril: the scale parameter of the distribution, a D x D lower-triangular matrix.
+    """
+    arg_constraints = {'loc': constraints.real, 'scale_tril': constraints.lower_triangular}
+    support = constraints.real
+    has_rsample = True
+
+    def __init__(self, loc, scale_tril, validate_args=None):
+        base_dist = Cauchy(loc, 1.0).to_event(1)
```

Comment:
Whoa, are you sure you want
```py
Transformed(Cauchy(loc, ...), LowerCholeskyAffine(0, scale_tril))
```
I would think you'd want to match `MultivariateNormal` and put the loc in the transform
```py
TransformedDistribution(Cauchy(torch.zeros_like(loc), 1.).to_event(1),
                        LowerCholeskyAffine(loc, scale_tril))
```

Reply:
Otherwise you'll get weird coupling of the `loc` and `scale_tril` parameters, right?

Reply:
maybe can you please explain? obviously math is the same...

Reply:
Hmm I am pretty sure the math is different:
```py
# This is the sampler you implemented:
scale_tril @ (noise + loc)

# This is the sampler I am suggesting:
scale_tril @ noise + loc
```
Which one of us is confused 😕 ?

Reply:
not related but may I ask a question here:

Can I use this LowerCholeskyAffine to define the MVT implementation at [this PR](https://github.com/pyro-ppl/pyro/pull/2025/files#diff-35ab0c5d82928aaf937d461c83519de2)? I got some trouble with writing a test to verify that the implementation is correct (I only verify the consistency with univariate case). I have a feeling that it might not be the same: MVT(df, loc, scale_tril) and TransformedDist(ST(df, 0, 1).expand([1])), Affine(loc, scale_tril))...

Reply:
@fritzo ah yeah you're right of course thanks for catching! 😊

Reply:
@fehiepsi you need to include the gamma variate outside the transform: z = mu + L z_unit / tau.sqrt() (or something like that)

Reply:
Thank Martin! So they are different. I'll think more about how to write a proper test for it when revisiting that PR.

Reply:
Oh the second thought, I think the implementation here uses a different definition w.r.t. the one in [wikipedia](https://en.wikipedia.org/wiki/Cauchy_distribution#Multivariate_Cauchy_distribution). Consider `loc=0`, `scale_tril=id`, log_prob of MVC here is `dist.Cauchy(0, 1).log_prob(x).sum(-1)` so it has the form of `log(1 + x ** 2).sum(-1)`, while the wikipedia definition has the form of `log(1 + ||x|| ** 2)`... If so, I think we should rename this distribution to something else.

Reply:
@fehiepsi yes Martin and I were discussing radial symmetry offline. I agree `1 + ||x||**2` seems more parsimonious.